### ISSUE TITLE
Tensorrt cannot appear on the webui

### ISSUE THREAD

**Issue by GitGod76** [AUTOMATIC1111/stable-diffusion-webui-tensorrt#7]
I get this error

Error loading script: [trt.py](https://trt.py/)

Traceback (most recent call last):

File "F:\stable-diffusion-webui\modules\[scripts.py](https://scripts.py/)", line 263, in load_scripts

script_module = script_loading.load_module(scriptfile.path)

File "F:\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module

module_spec.loader.exec_module(module)

File "<frozen importlib._bootstrap_external>", line 883, in exec_module

File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed

File "F:\stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt\scripts\[trt.py](https://trt.py/)", line 6, in <module>

from modules import script_callbacks, sd_unet, devices, shared, paths_internal

ImportError: cannot import name 'sd_unet' from 'modules' (unknown location)

**Comment by jilek-josef** [comment]
If it writes something about pycuda too than first install https://visualstudio.microsoft.com/cs/visual-cpp-build-tools/ so pycuda can be installed. Also you need to switch to dev branch for webui by: git checkout dev

**Comment by GitGod76** [comment]
I will try it, thank you

**Comment by fredrickflower** [comment]
Same error here as well

**Comment by MoreColors123** [comment]
> If it writes something about pycuda too than first install https://visualstudio.microsoft.com/cs/visual-cpp-build-tools/ so pycuda can be installed. Also you need to switch to dev branch for webui by: git checkout dev

thank you. when installing build tools, what needs to be enabled? there is a ton of options there.

**Comment by dmytropjsn** [comment]
This has nothing to do with pycuda or cpp build tools. The problem here is that it tries to import sd_unet from modules and this file appears to have been deleted for some odd reason.

For now, just checkout to commit 339b5315 and use it for now. It seems like automatic decided to purge the modules a bit and completely forgot that they made this extension which relied on sd_unet.

**Comment by jilek-josef** [comment]
Just like this
![10262](https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/assets/83363948/a51dfbe9-7d45-4037-837e-ba4ad777c4a8)

**Comment by MoreColors123** [comment]
thank you, this was exactly what was needed. it installs fine without any error now, on my side

**Comment by fredrickflower** [comment]
Any fix for "Error loading script: [trt.py](https://trt.py/)" then?

**Comment by AugmentedRealityCat** [comment]
Here is what I did to get pycuda to work. I am not saying YOU should do that. I am sure there are steps that are not necessary, and some might even break things that are currently working. But I'll share my experience and you do what you want with it, at your own risk. This is all on Windows 10, using the latest DEV branch.

I had first installed the Visual Studio C++ build tools but it was version 2022, which is the one offered by default by Microsoft. It was possible to include all the packages required, it seems, but it was not enough to get the whole thing to work. **So I removed Visual C++ Build Tools 2022.**

After removing version 2022, **I installed the version shown in the screenshot above, Visual C++ Build Tools** **2019**. I selected exactly the packages required, but I would still get errors about nvcc.exe and path and Cuda (if I remember correctly). But it was different and it looked like I was progressing.

I was about to change the path invormation linking to CUDA in the environment variables when I discovered that nvcc.exe and cuda were missing. I did a system restore yesterday to try to fix things, and maybe that's why it was gone. I don't know. Anyways, without it installed, I could not link to it in the environment variables. So I tried to find how to install nvcc.exe and cuda, which was to install the CUDA toolkit from Nvidia. And I remembered to check which version was compatible with Torch 2.0, so v 11.8.

I then installed that version of Cuda (which has nvcc.exe). It is important to note that CUDA was installed AFTER installing the Microsoft Visual C++ Build Tools 2019 with all the required packages. The version **I installed was cuda_11.8.0_522.06_windows**.

And only after that could I get the extension to install properly, and its tab to appear, and the whole thing to work without error messages.

Now, I managed to convert a safetensor into a onnx, and that onnx into a trt, and I got over 60 it/s when generating images with it after, while normally I get 30 it/s max in the best case scenario. 

I hope my little story will be helpful for someone ! (but use at your own risk - I am NOT a programmer, far from it)

TLDR:
1. removed Visual C++ build tools 2022
2. installed Visual C++ build tools 2019 + required packages
3. installed Nvidia Cuda Toolkit v11.8 (which works with Torch 2)
4. installed the extension properly without errors after that

**Comment by reallybigname** [comment]
Verified. I had to update Visual Studio Build Tools before it would install pycuda and work. I'm using Build Tools 2022.

**Comment by darongma** [comment]
> Here is what I did to get pycuda to work. I am not saying YOU should do that. I am sure there are steps that are not necessary, and some might even break things that are currently working. But I'll share my experience and you do what you want with it, at your own risk. This is all on Windows 10, using the latest DEV branch.
> 
> I had first installed the Visual Studio C++ build tools but it was version 2022, which is the one offered by default by Microsoft. It was possible to include all the packages required, it seems, but it was not enough to get the whole thing to work. **So I removed Visual C++ Build Tools 2022.**
> 
> After removing version 2022, **I installed the version shown in the screenshot above, Visual C++ Build Tools** **2019**. I selected exactly the packages required, but I would still get errors about nvcc.exe and path and Cuda (if I remember correctly). But it was different and it looked like I was progressing.
> 
> I was about to change the path invormation linking to CUDA in the environment variables when I discovered that nvcc.exe and cuda were missing. I did a system restore yesterday to try to fix things, and maybe that's why it was gone. I don't know. Anyways, without it installed, I could not link to it in the environment variables. So I tried to find how to install nvcc.exe and cuda, which was to install the CUDA toolkit from Nvidia. And I remembered to check which version was compatible with Torch 2.0, so v 11.8.
> 
> I then installed that version of Cuda (which has nvcc.exe). It is important to note that CUDA was installed AFTER installing the Microsoft Visual C++ Build Tools 2019 with all the required packages. The version **I installed was cuda_11.8.0_522.06_windows**.
> 
> And only after that could I get the extension to install properly, and its tab to appear, and the whole thing to work without error messages.
> 
> Now, I managed to convert a safetensor into a onnx, and that onnx into a trt, and I got over 60 it/s when generating images with it after, while normally I get 30 it/s max in the best case scenario.
> 
> I hope my little story will be helpful for someone ! (but use at your own risk - I am NOT a programmer, far from it)
> 
> TLDR:
> 
> 1. removed Visual C++ build tools 2022
> 2. installed Visual C++ build tools 2019 + required packages
> 3. installed Nvidia Cuda Toolkit v11.8 (which works with Torch 2)
> 4. installed the extension properly without errors after that

Thanks for this info. It helps me and I finally managed to get it working.

So, I follow direction to have the extension install on \stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt
Then, I extract the nvidia stuff and put it into \stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt\TensorRT-8.6.1.6

Then, I launch webui-user.bat and it give me a bunch of errors about not able to install pycuda.

To fix all that, I just install the following, then everything works. 

1.  Install Visual C++ build tools 2022
2.  Install Nvidia Cuda Toolkit v11.8

**Comment by stablegeniusdiffuser** [comment]
> So, I follow direction to have the extension install on \stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt
Then, I extract the nvidia stuff and put it into \stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt\TensorRT-8.6.1.6
>
> Then, I launch webui-user.bat and it give me a bunch of errors about not able to install pycuda.
>
> To fix all that, I just install the following, then everything works.
>
> Install Visual C++ build tools 2022
>Install Nvidia Cuda Toolkit v11.8

Thanks for trying to help - but I did exactly this and it still isn't working for me. Here's the relevant part of the webui start log:

```
Installing TensorRT wheel
Installing pycuda
Installing onnx

Launching Web UI with arguments: --xformers --autolaunch --disable-safe-unpickle --ckpt-dir C:\Temp\stable diffusion stuff\models
2023-06-05 18:18:35,732 - ControlNet - INFO - ControlNet v1.1.217
ControlNet preprocessor location: C:\Temp\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2023-06-05 18:18:35,810 - ControlNet - INFO - ControlNet v1.1.217
Error loading script: trt.py
Traceback (most recent call last):
  File "C:\Temp\stable-diffusion-webui\modules\scripts.py", line 256, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "C:\Temp\stable-diffusion-webui\modules\script_loading.py", line 11, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Temp\stable-diffusion-webui\extensions\stable-diffusion-webui-tensorrt\scripts\trt.py", line 6, in <module>
    from modules import script_callbacks, sd_unet, devices, shared, paths_internal
ImportError: cannot import name 'sd_unet' from 'modules' (unknown location)
```

**Comment by reallybigname** [comment]
Yep, has nothing to do with deforum.

**Comment by yuvraj108c** [comment]
Anyone managed to install pycuda on linux??

**Comment by cheadrian** [comment]
`***   *** WARNING: nvcc not in path.

***   *** May need to set CUDA_INC_DIR for installation to succeed.`

Installing [Cuda Toolkit V11.8](https://developer.nvidia.com/cuda-11-8-0-download-archive?target_os=Windows&target_arch=x86_64&target_version=11&target_type=exe_local) fixed this issue.

Also, I have Visual C++ build tools 2022 and 2019 installed. Don't think both are necessary.  

[<img src="https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/assets/20714227/91e7419f-36c7-4267-ba83-5714eb53431a" width="250"/>](https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/assets/20714227/91e7419f-36c7-4267-ba83-5714eb53431a) [<img src="https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/assets/20714227/17c41f4c-8ce8-4cdf-9cac-4c6772b02ca9" width="250"/>](https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/assets/20714227/17c41f4c-8ce8-4cdf-9cac-4c6772b02ca9)

Windows 11 / 10 SDK should not be necessary.
